src/Altinn.AccessManagement.Core/Services/SingleRightsService.cs

```diff
@@ -149,11 +158,90 @@ public async Task<DelegationActionResult> DelegateRights(int authenticatedUserId
         }
 
         /// <inheritdoc/>
-        public Task<List<Delegation>> GetOfferedRightsDelegations(AttributeMatch party)
+        public async Task<IEnumerable<RightDelegation>> GetOfferedRightsDelegations(AttributeMatch partyAttribute, CancellationToken cancellationToken = default)
         {
-            throw new NotImplementedException();
+            var delegations = await GetActiveOfferedRightsDelegationFromRepository(partyAttribute, cancellationToken, (delegation) => delegation?.ResourceType != "MaskinportenSchema");
+            return await ParseDelegationChanges(partyAttribute, delegations);
         }
 
+        private async Task<List<RightDelegation>> ParseDelegationChanges(AttributeMatch offeredBy, IEnumerable<DelegationChange> delegations)
+        {
+            var result = new List<RightDelegation>();
+            var resources = await _contextRetrievalService.GetResourceList();
+            foreach (var delegation in delegations)
+            {
+                var entry = new RightDelegation();
+                if (delegation.CoveredByPartyId != null)
+                {
+                    entry.To.Add(new()
```

Comment:
To/From: Needs party/userprofile lookup to set external identifiers (OrgNo, SSN or Username if an enterpriseuser)